oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/TokenHandler.java

```
@@ -233,6 +240,46 @@ public APIGatewayProxyResponseEvent tokenRequestHandler(APIGatewayProxyRequestEv
+            Optional<AuthCodeExchangeData> orchAuthCodeExchangeDataMaybe =
+                    segmentedFunctionCall(
+                            "orchAuthCodeService",
+                            () ->
+                                    orchAuthCodeService.getExchangeDataForCode(
+                                            requestBody.get("code")));
+
+            // TODO: ATO-1205: Remove these logs after consistency checks are complete.
+            if (orchAuthCodeExchangeDataMaybe.isEmpty()) {
+                LOG.warn(
+                        "OrchAuthCode consistency check error: Record present in Redis but not in DynamoDB. ClientId: {}. ClientSessionId: {}. NOTE: Redis is still the primary at present.",
+                        authCodeExchangeData.getClientId(),
+                        authCodeExchangeData.getClientSessionId());
+            } else {
+                AuthCodeExchangeData orchAuthCodeExchangeData = orchAuthCodeExchangeDataMaybe.get();
+
+                LOG.info(
+                        "OrchAuthCode consistency check: AuthCode and OrchAuthCode client ID equal? {}",
+                        authCodeExchangeData
+                                .getClientId()
+                                .equals(orchAuthCodeExchangeData.getClientId()));
+                LOG.info(
+                        "OrchAuthCode consistency check: AuthCode and OrchAuthCode client session ID equal? {}",
+                        authCodeExchangeData
+                                .getClientSessionId()
+                                .equals(orchAuthCodeExchangeData.getClientSessionId()));
+                LOG.info(
+                        "OrchAuthCode consistency check: AuthCode and OrchAuthCode auth time equal? {}",
+                        authCodeExchangeData
+                                .getAuthTime()
+                                .equals(orchAuthCodeExchangeData.getAuthTime()));
+            }
+        } catch (OrchAuthCodeException e) {
```

Are there any unchecked exceptions this might throw?

---

I think most are caught within `orchAuthCodeService.getExchangeDataForCode` and rethrown as `OrchAuthCodeException`

Happy to add a catch all block for the moment though thanks - will push that in a minute

---

I think most are caught within `orchAuthCodeService.getExchangeDataForCode` and rethrown as `OrchAuthCodeException`

I've updated the catch block to catch all exceptions during these consistency checks

Also added a todo comment to update this exception handling when we remove the write to Redis as we'll want to fail in a similar way to that when DynamoDB is the only store

---

Resolving (ran past Ethan)